pkg/operator/controllers/clusteroperatoraro/clusteroperatoraro_controller.go

```diff
@@ -94,7 +94,15 @@ func (r *Reconciler) setClusterOperatorStatus(ctx context.Context, originalClust
 		Reason:             reasonAsExpected,
 	})
 
-	if equality.Semantic.DeepEqual(clusterOperatorObj.Status.Conditions, originalClusterOperatorObj.Status.Conditions) {
+	clusterOperatorObj.Status.Versions = []configv1.OperandVersion{
+		{
+			Name:    "operator",
+			Version: version.GitCommit,
+		},
+	}
+
+	if equality.Semantic.DeepEqual(clusterOperatorObj.Status.Conditions, originalClusterOperatorObj.Status.Conditions) &&
+		equality.Semantic.DeepEqual(clusterOperatorObj.Status.Versions, originalClusterOperatorObj.Status.Versions) {
```

Comment:
what's the purpose of the `equality` deepequal compared to the `reflect` one? (it's probably valid but i'm still curious)

Reply:
I originally added the first check when handling operator conditions. I lifted the comparison from [openshift/library-go](https://github.com/openshift/library-go/blob/d7e4a8ca058837a972a4cdef06334be4f7becdde/pkg/operator/status/status_controller.go#L208)'s equivalent status controller, whose helper methods I used directly but we did not use their controller itself.

Reply:
Looking at [the implementation](https://github.com/kubernetes/apimachinery/blob/v0.27.1/pkg/api/equality/semantic.go), it appears to just be a wrapper of (a fork of) golang reflect.DeepEqual with some thoughtful k8s-specific equality functions provided.